swap/swap.go

```diff
@@ -371,6 +396,47 @@ func (s *Swap) handleEmitChequeMsg(ctx context.Context, p *Peer, msg *EmitCheque
 	return err
 }
 
+func (s *Swap) handleConfirmChequeMsg(ctx context.Context, p *Peer, msg *ConfirmChequeMsg) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	cheque := msg.Cheque
+
+	if p.getPendingCheque() == nil {
+		p.logger.Warn("ignoring confirm msg, no pending cheque", "got", cheque)
+		return
+	}
+
+	if !cheque.Equal(p.getPendingCheque()) {
+		p.logger.Warn("ignoring confirm msg, wrong cheque", "got", cheque, "expected", p.getPendingCheque())
+		return
+	}
+
+	err := p.setLastSentCheque(cheque)
+	if err != nil {
+		p.Drop("persistence error")
+		return
+	}
+
+	err = p.setPendingCheque(nil)
+	if err != nil {
+		p.Drop("persistence error")
+		return
+	}
+
+	err = p.updateBalance(int64(cheque.Honey))
+	if err != nil {
+		p.Drop("persistence error")
+		return
+	}
+
+	// since more swap traffic might have occurred since this cheque was already sent, we redo the payment threshold check
+	err = s.checkPaymentThresholdAndSendCheque(p)
+	if err != nil {
+		p.logger.Warn("failed to send already due cheque", "error", err)
+		return
```

Comment:
this is probably for another PR, but we should do something here right?

Reply:
I don't think there's much more we can do here other than logging. Sending will be tried again the next time we are in `Add`.